Update the docker images to run as yuzu user (UID 1027)

jroweboy opened the pull request:
This changes the script to use the `yuzu` user for anyone that runs the image. As a docker best practice, containers should try to only run as an unprivileged user https://docs.docker.com/engine/security/userns-remap/ to prevent privilege escalation attacks.

Also fixed a few small issues that crept in in the previous commits.

DarkLordZach commented:
I am pretty sure that this PR isn't necessary for our use case of docker.
The docker best practices are written with applications in mind, not build images. Many of the guidelines are very solid and apply to both, but some are either detrimental or unnecessary for build images.

One of them is worrying about privilege escalation. For one, we are not running the images on our own infrastructure, its being run on Azure. I am confident Azure has protections in place so that we do not need to worry. Additionally, this is targeted more towards runtime/runner containers, where the commands and code provided is more user-controlled. In our case, all of the code exec'd in these images is controlled by us.

Ultimately, while this isn't harmful, per say, I wouldn't say its necessary either. It also has the potential for problems when root permissions are taken for granted.

jroweboy commented:
@DarkLordZach I agree that its not strictly necessary, but we were kinda abusing root permission in the docker image to hack around things instead of fixing the issue in a more sustainable way. For example in the mingw container build script, we were doing the following https://github.com/yuzu-emu/yuzu/pull/3735/commits/a147d652ea890658bd158bc04f158031db75a3d5#diff-5cee4a0ed57ee784e9f20a24e7281b39L7 before this, which required root permissions. This made it so when someone tried to build with a cross mingw compiler on their system without docker, they couldn't replicate the mingw build. And yes, we've had a few people recently asking me why mingw cross doesn't build on their system which is why i removed those hacks and fixed it in a different way here: https://github.com/yuzu-emu/yuzu/pull/3735/commits/a147d652ea890658bd158bc04f158031db75a3d5#diff-af3b638bc2a3e6c650974192a53c7291R365

And yes, the builds are running on Azure, but I don't think we should take that for granted. We are already pretty limited in runtime minutes on Azure, so I wouldn't be surprised if someone moves us to Github Actions in the near future, and we start running our own github-runner. So we shouldn't rely on host system sandboxing.
